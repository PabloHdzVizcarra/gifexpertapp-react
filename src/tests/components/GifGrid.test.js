import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('pruebas en <GifGrid />', () => {

  const category = 'Happy';
  
  test('el componente debe renderizarse correctamente', () => {
    
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })

    const wrapper = shallow(<GifGrid category={category}/>);
    expect(wrapper).toMatchSnapshot();
  })

  test('debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
    
    const gifs = [
      {
        id: 'ABC',
        url: 'http://text.gif',
        title: 'cualquier img'
      },
      {
        id: 'XYZ',
        url: 'http://text.gif',
        title: 'cualquier img'
      }
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    })

    const wrapper = shallow(<GifGrid category={category} />);
    
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    //console.log(wrapper.find('GifGridItem').length)
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

  })
  
  
})

