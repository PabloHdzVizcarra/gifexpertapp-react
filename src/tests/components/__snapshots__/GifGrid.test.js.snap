// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGrid /> debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Happy
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier img"
      url="http://text.gif"
    />
    <GifGridItem
      id="XYZ"
      key="XYZ"
      title="cualquier img"
      url="http://text.gif"
    />
  </div>
</Fragment>
`;

exports[`pruebas en <GifGrid /> el componente debe renderizarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Happy
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
